src/App.tsx

```diff
@@ -10,33 +12,29 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTabId] = useState('tab-1');
+
+  const currentTab = tabs.find(tab => tab.id === selectedTabId) || tabs[0];
+
+  const onTabSelected = (tab: Tab) => {
+    if (tabs.find(someTab => someTab.id === tab.id)) {
+      setSelectedTabId(tab.id);
+    } else {
+      setSelectedTabId(tabs[0].id);
```

Comment:
You already check it when finding a tab, so there is actually no need to double check